pkg/thing/interactors/register_thing.go

```diff
@@ -45,28 +45,24 @@ func (i *ThingInteractor) Register(authorization, id, name string) error {
 	i.logger.Debug("Executing register thing use case")
 	err := i.verifyThingID(id)
 	if err != nil {
-		errReply := i.reply(id, "", err)
-		if errReply != nil {
-			return fmt.Errorf("error sending success response to client: %v: %w", errReply, err)
-		}
-		return fmt.Errorf("error registering thing: %w", err)
+		sendErr := i.sendResponse(id, "", err)
+		return fmt.Errorf("error registering thing: %w", sendErr)
 	}
 
 	// Get the id generated as a token and send in the response
 	token, err := i.thingProxy.Create(id, name, authorization)
-	errReply := i.reply(id, token, err)
+	sendErr := i.sendResponse(id, token, err)
 	if err != nil {
-		if errReply != nil {
-			return fmt.Errorf("error sending success response to client: %v: %w", errReply, err)
-		}
-		return fmt.Errorf("error registering thing: %w", err)
-	}
-	if errReply != nil {
-		i.logger.Error(errReply)
+		// was not possible to create a thing, so returns without send request to connector
+		return fmt.Errorf("error registering thing: %w", sendErr)
 	}
 
 	err = i.connectorPublisher.SendRegisterDevice(id, name)
 	if err != nil {
+		if sendErr != nil {
```

Comment:
Isn't this error handled already on the `sendResponse()` method?

Reply:
@joaoaneto Yes, it is handled in sendResponse, but it still needs to be returned in the case `sendErr != nil`. However, in line 35, the `sendError` should be returned instead of `nil`